integration-test/src/main/java/uk/gov/gchq/gaffer/integration/junit/extensions/IntegrationTestSuiteExtension.java

```diff
@@ -0,0 +1,320 @@
+     * @return a {@link ConditionEvaluationResult#enabled(String)} or
+     * {@link ConditionEvaluationResult#disabled(String)} {@link Object} for the
+     * {@link org.junit.jupiter.api.Test} in question
+     */
+    @Override
+    public ConditionEvaluationResult evaluateExecutionCondition(final ExtensionContext context) {
+        if (context.getTestMethod().isPresent()) {
+            final String currentMethodName = context.getTestMethod().get().getName();
+            if (this.skipTestMethods.containsKey(currentMethodName)) {
+                return ConditionEvaluationResult.disabled(this.skipTestMethods.get(currentMethodName));
+            }
+        }
+        return ConditionEvaluationResult.enabled("Test enabled");
+    }
+
+    private static Set<Object> getSuiteObjects(final IntegrationTestSuite integrationTestSuite) {
+        final Set<Object> objects = integrationTestSuite.getObjects().orElse(Collections.emptySet());
+        LOGGER.debug("Retrieved the following objects types from the IntegrationTestSuite: [{}]",
+                objects.stream().map(o -> o.getClass().getName()).collect(Collectors.toList()));
+        return objects;
+    }
+
+    private Map<String, String> getSkipTestMethods(final IntegrationTestSuite integrationTestSuite) {
+        final Map<String, String> skipTestMethods = integrationTestSuite.getTestsToSkip().orElse(Collections.emptyMap());
+        LOGGER.debug("Retrieved the following skip-test methods from the IntegrationTestSuite: [{}]",
+                StringUtils.join(skipTestMethods));
+        return skipTestMethods;
+    }
+
+    private void injectInstanceFields(final Object instance) {
+        findAnnotatedFields(instance.getClass(), IntegrationTestSuiteInstance.class, ReflectionUtils::isNotStatic).forEach(field -> {
+            try {
+                LOGGER.debug("Field [{}] requires injecting", field);
+                final Object object = getObject(field.getType());
+                LOGGER.debug("Object [{}] found for the field", object);
+                makeAccessible(field).set(instance, object);
+            } catch (final Throwable t) {
+                ExceptionUtils.throwAsUncheckedException(t);
+            }
+        });
+    }
+
+    private Object getObject(final Class<?> type) {
+        final Set<Object> set = suiteCache.stream()
+                .filter(o -> type.isAssignableFrom(o.getClass()))
+                .collect(Collectors.toSet());
+        if (!set.isEmpty()) {
+            return set.iterator().next();
+        } else {
+            throw new ParameterResolutionException(String.format("Object of type [%s] not found", type));
+        }
+    }
+
+    private static IntegrationTestSuite getIntegrationTestSuite(final String initClass) {
+        final IntegrationTestSuite integrationTestSuite;
+        if (INTEGRATION_TEST_SUITE_CLASS_MAP.containsKey(initClass)) {
+            integrationTestSuite = INTEGRATION_TEST_SUITE_CLASS_MAP.get(initClass);
```

Comment:
Can you explain why not?

Reply:
Not sure how to do this... I can run the test suite twice but not sure of a way prove that the condition block has been entered. I'll investigate...

Reply:
I would have thought it could be tested by using the same 'initClass' twice. Wouldn't this happen if the suite selects two (or more) different classes? I think all the existing Suites in the unit tests have just one `@SelectClasses`. Running the suite twice wouldn't trigger this as everything will be torn down in between the runs.

Reply:
Yeah, but how do you check that this block is actually used other than looking at the report after? That's what I'm looking into

Reply:
Found a way of proving it in a roundabout way but it is a bit convoluted. See what you think...

Reply:
Had another go today and made it a lot simpler!